### DISCUSSION TITLE
Azure NFS Share File AKS mount options

### DISCUSSION THREAD

**Discussion by jcamu** [kubernetes-sigs/azurefile-csi-driver#920] [Q&A] [answered]
Hello,

I want to be able to add mountOptions on persistentVolume using the file.csi.azure.com to be able to mount a shared and static File Share.

I see that for dynamic provisoning there is such options in storageClass resource, but for static provisionning i don't find it. (When i try to put some uid=0, it not works)
In addition the uid used when creating a new file in the NFS is the one from the Admin cluster which have create the PV and PVC.

Did i miss something? Is it not possible?

Regards

**Comment by andyzhangx** [comment, marked as answer]
`uid=xx` is not a valid mount options for nfs mount, are you seeking this `securityContext` for pod?

https://github.com/kubernetes-sigs/azurefile-csi-driver/blob/a82ae49a03908b7f97896c0db8e721022b386bc1/deploy/example/statefulset-nonroot.yaml#L17-L20

**Reply by jcamu** [reply]
Ok thanks for the answer.
I suppose in the same order, if there is no documentation on secret management for NFS it's because the only way to secure Azure NFS File share is by the use of private endpoint and limit access by subnet  no?  (No secret like with File Share)

**Reply by andyzhangx** [reply]
correct, secret is only for smb share, and for nfs, it could only be accessed by same vnet.
